plugin/src/init.server.lua

```diff
@@ -1,222 +1,278 @@
-	local tree = encodeInstance(game, filterServices)
+local function SendDataModelInfo()
+	local Tree = EncodeInstance(game, FilterServices)
 
-	local success, result = pcall(HttpService.RequestAsync, HttpService, {
+	local Success, Result = pcall(HttpService.RequestAsync, HttpService, {
 		Method = "POST" :: "POST",
-		Url = string.format("http://localhost:%s/full", Settings.port),
+		Url = string.format("http://localhost:%s/full", tostring(Settings.Port)),
 		Headers = {
 			["Content-Type"] = "application/json",
 		},
 		Body = HttpService:JSONEncode({
-			tree = tree,
+			tree = Tree,
 		}),
+
 		Compress = Enum.HttpCompression.Gzip,
 	})
 
-	if not success then
-		warn(`[Luau Language Server] Connecting to server failed: {result}`)
-		cleanup()
-	elseif not result.Success then
-		warn(`[Luau Language Server] Sending full DM info failed: {result.StatusCode}: {result.Body}`)
-		cleanup()
+	if not Success then
+		warn(`[Luau Language Server] Connecting to server failed: {Result}`)
+		CleanUpConnections()
+	elseif not Result.Success then
+		warn(`[Luau Language Server] Sending DataModel info failed: {Result.StatusCode}: {Result.Body}`)
+		CleanUpConnections()
 	else
-		connected.Value = true
-		if not isSilent then
-			print(`[Luau Language Server] Successfully sent full DataModel info`)
-		end
+		Connected = true
+		ConnectButton.Icon = "rbxassetid://11116536087"
+		warn("[Luau Language Server] Now sending DataModel information.")
 	end
 end
 
-local function watchChanges(isSilent)
-	local sendTask: thread?
-	if connected.Value or Settings == nil then
-		if not isSilent then
-			warn("[Luau Language Server] Connecting to server failed: invalid settings")
-		end
+local function WatchChanges()
+	local SendTask: thread?
+	if Connected or Settings == nil then
+		warn("[Luau Language Server] Connecting to server failed: Already connected, or settings non-existent.")
 		return
 	end
-	cleanup()
+	CleanUpConnections()
 
-	local function deferSend()
-		if sendTask then
-			task.cancel(sendTask)
+	local function DeferSendDataModelInfo()
+		if SendTask then
+			task.cancel(SendTask)
 		end
-		sendTask = task.delay(0.5, function()
-			sendFullDMInfo(isSilent)
-			sendTask = nil
+		SendTask = task.delay(0.5, function()
+			SendDataModelInfo()
+			SendTask = nil
 		end)
 	end
 
-	-- TODO: we should only send delta info if possible
-	local function descendantChanged(instance: Instance)
-		for _, service in Settings.include do
-			if instance:IsDescendantOf(service) then
-				deferSend()
-				return
+	for _, Service: Instance in Settings.Include do
+		table.insert(
+			Connections,
+			Service.DescendantAdded:Connect(function(Descendant: Instance)
+				if table.find(CLASSES_EXCEPTION_LIST, Descendant.ClassName) ~= nil then
+					return
+				end
+				--print(`[Luau Language Server] {Descendant.Name} added to {Service.ClassName}`)
+				table.insert(
+					Connections,
+					Descendant.Changed:Connect(function(Property: string)
```

Comment:
how does Changed perform on large places ? and why we send dmInfo on all the changes when we can just track the name with `GetPropertyChangedSignal("Name")` , we only need name for the dminfo

Reply:
Addressed in another comment

Reply:
No performance gripes on medium sized place from me, however can agree that other properties are not neccesary, DescendantAdded handles the Parent, and then Name is the only neccesary field. That should then address potential performance issues with property changes on all descendants, since Name changing will most definitely not cause problems.

Will also be able to remove CLASSES_EXCEPTION_LIST and related functionality, meaning no need to implement one of the future features I mentioned